Scripts/lifeCycleAppObject.js

```diff
@@ -97,10 +97,13 @@ class appConfig {
             this.history.push(this.state);
             logger.info('New appState pushed to history: ' + newState);
           }
-          if (!stateTransition.includes(newState)) {
+          if (!stateTransition.includes(newState) && currentState.state != newState) {
             cy.log('Requested state transition for application is not supported');
             this.state = currentState;
           }
+          if (currentState.state == newState) {
+            this.state = currentState;
+          }
         }
 
```

Comment:
why do we need this change here ? expecting changes only inside **setAppState** function

Reply:
I think the above change is also required because the base logic here is if the newState is not present in 'allowedStateTransitions' config, then control comes to if condition in line 101, where it prints the log `cy.log('Requested state transition for application is not supported')` and assigns the state object to currentstate as there is no transition to newState. So since the same app transitions are also not part of 'allowedStateTransitions' config, say if u look at the relaunch scenario logs for example, you can see the same log. But ideally that's not correct right?
We can't say same state transition is not supported right as that could be misleading, so basically we are just  removing printing the log for same state transitions now that we have added action for it, rest is same. So its just a minor change to avoid printing the log and nothing major